src/preferences/dialog/dlgprefdeck.cpp

```diff
@@ -495,11 +513,31 @@ void DlgPrefDeck::slotRateRampingModeLinearButton(bool checked) {
     }
 }
 
+void DlgPrefDeck::slotTimeFormatChanged(double v) {
+    int i = static_cast<int>(v);
+    m_timeDisplayFormat = static_cast<TrackTime::DisplayFormat>(i);
+
+    m_pConfig->set(ConfigKey("[Controls]","TimeFormat"), ConfigValue(v));
+    comboBoxTimeFormat->setCurrentIndex(
+                comboBoxTimeFormat->findData(i));
+
+}
+
+void DlgPrefDeck::slotTimeFormatChanged(int index) {
+    int mode = comboBoxTimeFormat->itemData(index).toInt();
+    m_timeDisplayFormat = static_cast<TrackTime::DisplayFormat>(mode);
```

Comment:
I think you can omit the m_timeDisplayFormat here and read the comboBoxTimeFormat directly